src/update/UpdateManager.cpp

```diff
@@ -159,8 +171,8 @@ void UpdateManager::GetUpdates(bool ignoreNetworkErrors)
                                   ErrorDialogOptions{ ErrorDialogType::ModalErrorReport });
                                });
                             }
-                            
-                            mUpdateMutex.unlock();
+
+                            mOnProgress = false;
```

Comment:
You reset mOnProgress to false on some of the exit paths.

But I think you must do it more places.

The returns at lines 124 and 139 -- where you detect a network failure or a corruption of the download -- should also reset mOnProgress to false.  Otherwise, automatic update checking will never be tried again, while the program remains running.

Reply:
But there is an even simpler, and preferable, thing you can write:

```
auto cleaup = finally([&]{ mOnProgress = false; });
```

Insert that line just after mOnProgress is made true, but also so that the end of the scope of `cleanup` is the correct place.

Then at one stroke, you cover all possible exit paths.  You miss none of them.

See other uses of `finally` in Audacity.

Reply:
And I think the body of the lambda in the `finally` should also do
```
wxTheApp->CallAfter([this]{ mProgressDialog.reset(); });
```

which would also easily remove doubts whether you are cleaning up the dialog on all possible paths, including error paths.

Reply:
Yes, because I add it manualy, therefore I agree with about RAII approach for fix the same possible problems.

Reply:
Also I think I solve this possible moments by create own `UdateLocker` class, that atomic blocking entry to update process with RAII support.

Reply:
Doing it with RAII would be even better.  See ValueRestorer in MemoryX.h which may already do what you want.  No new class needed.

```
ValueRestorer<bool> cleanup;
...
cleanup = valueRestorer(mOnProgress, true);  // inside the lock on the mutex
...
[cleanup = std::move(cleanup), ... // to transfer the cleanup from one lambda into another
```

Do you understand the idea?

Reply:
Yes, I implemented something like this, but thanks for `valueRestorer` example, I think I can change the part of same logic by `valueRestorer` using.